### ISSUE TITLE
[semi] Won't trigger on semicolons after overrides

### ISSUE THREAD

**Issue by nartallax** [typescript-eslint/typescript-eslint#4317]
- [+] I have tried restarting my IDE and the issue persists.
- [+] I have updated to the latest version of the packages.
- [+] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

Rule @typescript-eslint/semi (and @typescript-eslint/member-delimiter-style) seem to completely ignore semicolons after method override without implementation.

.eslintrc.js: 
```JS
module.exports = {
  root: true,
  parser: "@typescript-eslint/parser",
  plugins: [
    "@typescript-eslint",
  ],
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
  ],
  rules: {
    // ... other rules
    "semi": "off",
    "@typescript-eslint/member-delimiter-style": ["warn", {
      multiline: {delimiter: "none"}, 
      singleline: {delimiter: "comma", requireLast: false}
    }],
    "@typescript-eslint/semi": ["warn", "never"]
  }
}
```

Example code:
```TS
export class Cls {
	action(): void;
	action(value: string): void;
	action(value?: string): void {
		console.log(value)
	}
}
```

**Expected Result**

All the semicolons in the example code above should trigger warning.

**Actual Result**
No warning is produced for any of the semicolons.

**Additional Info**
No errors/exceptions appear in eslint logs, other rules work fine, these two rules also work fine in other cases.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.7.0` |
| `@typescript-eslint/parser`        | `5.7.0` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.4.1` |
| `node`                             | `16.13.0` |
